microcosm_pubsub/dispatcher.py

```diff
@@ -24,25 +25,46 @@ class SQSMessageDispatcher:
     Dispatch batches of SQSMessages to handler functions.
 
     """
+    logger: Logger
+
     def __init__(self, graph):
         self.opaque = graph.opaque
         self.sqs_consumer = graph.sqs_consumer
         self.sqs_message_context = graph.sqs_message_context
         self.sqs_message_handler_registry = graph.sqs_message_handler_registry
         self.send_metrics = graph.pubsub_send_metrics
+        self.send_batch_metrics = graph.pubsub_send_batch_metrics
         self.max_processing_attempts = graph.config.sqs_message_dispatcher.message_max_processing_attempts
 
     def handle_batch(self, bound_handlers) -> List[MessageHandlingResult]:
         """
         Send a batch of messages to a function.
 
         """
-        instances = [
-            self.handle_message(message, bound_handlers)
-            for message in self.sqs_consumer.consume()
-        ]
+        with elapsed_time(self.opaque):
+            instances = [
+                self.handle_message(message, bound_handlers)
+                for message in self.sqs_consumer.consume()
+            ]
+
+        message_count = len([
+            instance for instance in instances
+            if instance.result != MessageHandlingResultType.IGNORED
```

Comment:
I think knowing about ignored messages is potentially interesting too, somewhere. Just to make sure they aren't burning resources.

Reply:
I was thinking about that, actually. The problem then is sort of then you'd get a constant "10" here, and that's not super useful. I think it's a safe assumption (for now) that `IGNORED` messages are fast enough (since it should be a relegated to a single hash table lookup).